### PR TITLE
Two small performance fixes (reduce garbage creation)

### PR DESCRIPTION
**Pull request by rorygraves**

### CODE REVIEW

File: rsc/src/main/scala/rsc/scan/Scanner.scala [hunk 1]
```diff
@@ -447,8 +447,9 @@ final class Scanner private (
   }
 
   private def emitIdOrKeyword(): Unit = {
-    val token = keywords.getOrElse(lexeme, ID)
-    val value = if (token == ID) lexeme else null
+    val element = lexeme
```
Comment (rorygraves) [inline rsc/src/main/scala/rsc/scan/Scanner.scala]: each call to lexeme allocates a new String, separating this out halves the number of strings generates.

File: rsc/src/main/scala/rsc/typecheck/Typechecker.scala [hunk 2]
```diff
@@ -16,7 +16,7 @@ final class Typechecker private (
   def apply(env: Env, tree: Typeable): Type = {
     if (tree.pos != NoPosition) {
       val oldPos = reporter.pos
-      reporter.pos = tree.point
+      reporter.pos = treePoint(tree)
```
Comment (rorygraves) [inline rsc/src/main/scala/rsc/typecheck/Typechecker.scala]: This is a very hot path - each call to ```tree.point``` created an instance of TreePointOps.

**Comment by xeno-by** [comment]
Hi @rorygraves! We are super inspired by your excitement - you got everything working shortly after the release and jumped straight into hacking 👍 

Since Rsc is a performance-focused compiler, we would like to have all code changes accompanied by performance benchmarks. We'll soon start working on an infrastructure to do automatic pull request benchmarking (https://github.com/twitter/reasonable-scala/issues/3), but in the meanwhile, could you run `sbt bench` before and after the pull request and submit the results in a comment to this PR?

**Comment by rorygraves** [comment]
@xeno-by woot - I'm excited by performance too!. I will run the benchmarks and provide evidence (will try and do it this weekend).

**Comment by xeno-by** [comment]
@rorygraves Over the weekend, I've improved the benchmarking infrastructure (now, we have the `benchJVM` command that can be used for lightweight performance testing with reasonable run-to-run variation). Then, I started playing with your ideas as well as some of the ["Performance"](https://github.com/twitter/reasonable-scala/issues?q=is%3Aissue+is%3Aopen+label%3APerformance) tickets. 

Take a look at my current progress here: https://github.com/xeno-by/reasonable-scala/commits/topic/tuning. For example, the continuation of the Tree.point idea yielded [a ~25% speedup of HotRscTypecheck](https://github.com/xeno-by/reasonable-scala/commit/efe5a96c07b02813c924293030ab1f979fcfffb3)! In the coming days, I'll be taking a look at `lexeme` and the remaining tickets.

**Comment by rorygraves** [comment]
@xeno-by - Awesome stuff.  I have lots of small changes to investigate/test/PR.  This PR is now dead (I think you have taken both changes and applied them on your branch).
I've raised another PR for a change involving lexeme you didn't get ( https://github.com/twitter/reasonable-scala/pull/30 ) .

**Comment by rorygraves** [comment]
p.s. benchJVM is really useful

**Comment by rorygraves** [comment]
p.p.s 25% is amazing!

**Pull request closed without merging**